python/dgl/graphbolt/impl/gpu_graph_cache.py

```diff
@@ -0,0 +1,107 @@
+"""HugeCTR gpu_cache wrapper for graphbolt."""
+import torch
+
+
+class GPUGraphCache(object):
+    r"""High-level wrapper for GPU graph cache.
+
+    Places the GPU graph cache to torch.cuda.current_device().
+
+    Parameters
+    ----------
+    num_edges : int
+        Upperbound on number of edges to cache.
+    threshold : int
+        The number of accesses before the neighborhood of a vertex is cached.
+    indptr_dtype : torch.dtype
+        The dtype of the indptr tensor of the graph.
+    dtypes : list[torch.dtype]
+        The dtypes of the edge tensors that are going to be cached.
```

Comment:
Did you consider wrap the graph in the cache directly so that the caller of the cache does not need to worry about query / miss / replace?

Reply:
Yes, it needs to be seperate so that we can run the cache operations in a nonoverlapped manner while the fetch of missing vertices is performed in an overlapped manner.